src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLICATION_FOR_CAPACITY = 1.5;
+    private static final String EXCEPTION_MESSAGE = "Index of the bounds. Check your input index!";
+    private int size;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(EXCEPTION_MESSAGE);
+        }
+        if (size == items.length) {
+            grow();
+        }
+        System.arraycopy(items, index, items, index + 1, size - index);
+        items[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkRangeOfIndex(index);
+        return items[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkRangeOfIndex(index);
+        items[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRangeOfIndex(index);
+        T item = items[index];
+        System.arraycopy(items, index + 1, items, index, size - 1 - index);
+        size--;
+        return item;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == items[i] || element != null && element.equals(items[i])) {
```

Comment:
you twice do one thing, element == items[i], element.equals(items[i]). We can add null to ArrayList and we must remote null from ArrayList